src/oic/utils/http_util.py

```diff
@@ -284,29 +275,29 @@ def cookie_signature(key, *parts):
        :type parts: list of bytes or strings
        :returns: hexdigest of the HMAC
     """
-    assert isinstance(key, binary_type)
+    assert isinstance(key, bytes)
     sha1 = hmac.new(key, digestmod=hashlib.sha1)
     for part in parts:
         if part:
-            if isinstance(part, text_type):
+            if isinstance(part, str):
                 sha1.update(part.encode('utf-8'))
             else:
                 sha1.update(part)
-    return text_type(sha1.hexdigest())
+    return str(sha1.hexdigest())
 
 
 def verify_cookie_signature(sig, key, *parts):
     """Constant time verifier for signatures
 
        :param sig: The signature hexdigest to check
-       :type sig: text_type
+       :type sig: str
        :param key: The HMAC key to use.
        :type key: bytes
        :param parts: List of parts to include in the MAC
        :type parts: list of bytes or strings
        :raises: `InvalidCookieSign` when the signature is wrong
     """
-    assert isinstance(sig, text_type)
+    assert isinstance(sig, str)
     return safe_str_cmp(sig, cookie_signature(key, *parts))
```

Comment:
safe_str_cmp was only needed because ancient Py 2.7 versions have no hmac.compare_digest() for constant time comparision.

So this can be replaced by hmac.compare_digest().

Reply:
OK, I will probably change that as well.

I have later noticed some other checks for python version, so I give it another look.

We are not rushing to merge this, I still need to make the bugfix released before I am going to merge this anyway :)

Reply:
ok. Had the impression you were rather eager to merge it as soon as possible ;-).

Reply:
I want to merge before doing more changes to the codebase as it should make the work easier :)